src/store/reducers/appReducer.js

```diff
@@ -178,6 +214,13 @@ const updateTimeInterval = (state, { payload }) => update(state,
   { timeInterval: { $set: payload } }
 );
 
+const updateDiagnosticsOptOut = (state, { payload }) => update(state,
+  {
+    diagnosticsOptOut: {
+      $set: isDef(payload.diagnosticsOptOut) ? payload.diagnosticsOptOut : true
```

Comment:
We shouldn't need the ternary any more since the service call will handle the `undefined` case. Also instead of passing the entire response object to the reducer just pass the new value for the diagnosticsOptOut.

Reply:
done